File: Neetcode/Permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        output = []
        res = []
        def bt():
            if len(res) >= len(nums):
                output.append(res.copy())
                return
            for n in nums:
                if n not in res:
                    res.append(n)
                    bt()
                    res.pop()
        bt()
        return output

File: Neetcode/Permutations.cpp
class Solution {
public:
    vector<vector<int>> out;
    vector<int> res;
    void bt(vector<int> nums){
        if (res.size() == nums.size()){
            out.push_back(res);
            return;
        }
        for(int i=0; i<nums.size(); i++){
            if(find(res.begin(), res.end(), nums[i]) == res.end()){
                res.push_back(nums[i]);
                bt(nums);
                res.pop_back();
            }
        }
    }
    vector<vector<int>> permute(vector<int>& nums) {
        bt(nums);
        return out;
    }
};
